### ISSUE TITLE
Support for .NET Standard

### ISSUE THREAD

**Issue by Styxxy** [SimpleBrowserDotNet/SimpleBrowser#189]
Add support for .NET Standard. Probably will have to refactor some small stuff, to get "rid" of the dependency of RazorHosting dll.

This issue depends on #188 (upgrade to vs2017 and new csproj file format).

**Comment by kevingy** [comment]
I have some concerns here. Doing this without maintaining a parallel track will break everyone not working with .NET Standard. This project has been around for years. I don't want to orphan everyone using the current code base.

**Comment by Styxxy** [comment]
The great thing with the new csproj format, is that you can easily target multiple versions. In this case, I would target both net4 and netstandard (exact version to be yet determined; lowest version as possible though). The nuget package can also contain multiple versions (for net4 and netstandard). Depending on the project version of the consumer, Nuget will choose the appropriate dll from the package.

Note that netstandard is also supported in full .net (since 4.5 and higher). Also check out : https://docs.microsoft.com/en-us/dotnet/standard/net-standard

I'll work on this and will keep in mind backwards compatibility ;-).

**Comment by kevingy** [comment]
FYI: I'm looking at this pull request today. I'm sorry for the delay.

**Comment by Styxxy** [comment]
@kevingy 
Adding support for .NET Standard (2.0) is going well. However I still have a few questions.

The rendering is currently done by the library "RazorHosting". For netstandard I chose the library https://github.com/toddams/RazorLight for the moment. This will add many additional dependencies though (and I am still looking to get it completely working). If you wish to use another library, just let me know.
Personally though, I would rather refactor the whole log rendering. The main library should have interfaces to provide the "infrastructure" (targets) and actual targets (like html rendering by Razor) can be additional libraries that can be installed. However, this is quite a _breaking_ change to the current code base; but is also worth considering.

I upgraded the unit test library to .NET 4.5, as I also wanted to unit test the netstandard (using dotnetcore2). The main library is still .NET 4.0 (and .NET Standard 2).
I added one unit test for testing the custom headers. Also I disabled the currently failing unit test (because the external website that we use to test, changed). In the dotnetcore target I also disabled (condition!) the unit test for custom host header, because in dotnetcore the Host header is not added to the headers collection anymore (but it is being sent!). The "Host" property on the webrequest is being set correctly, but we can't really see this in the requestlog.
There are still a few other unit tests failing, but I still have to investigate them.
